### DISCUSSION TITLE
Whats the point of string type limitation with UseProperties for AdoJobStore?

### DISCUSSION THREAD

**Discussion by justmara** [quartznet/quartznet#1182] [Q&A] [answered]
Now when trying to use any typed UsingJobData method with `UseProperties=true` will throw 
> JobDataMap values must be strings when the 'useProperties' property is set.

But the property-setting factory itself handles all the primitive types well. So this `hack` works perfectly:
```csharp
var data = new Dictionary<string, object>();
data["IntProperty"] = 500;
job.UsingJobData(new JobDataMap(jobConfig.Data as IDictionary<string, object>));
```
It really sets the property defined with int type to 500 on this job
```csharp
public class TestJob : IJob {
    public int IntProperty { get; set; }
}
```
Under the hood the JobData uses `StringKeyDirtyFlagMap : DirtyFlagMap<string, object>`, which itself uses `Dictionary<string, object>` for key-value store. So whats the point of this limitation applied only to explicit typed overloads - i.e. `sugar`-methods? It does not make any sense in terms of functionality.

This check happens only in StdAdoDelegate's [ConvertToPropery](https://github.com/quartznet/quartznet/blob/99cfca10836a9e0868d74d7a5ccb687a650c1e92/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.cs#L540) method who's output's only meaning is to be serialized to `byte[]`. So only because of this IDictionary -> NameValueCollection intermediate convertion this check kicks in. And only for Ado storage.

The problem is that only this ConvertToPropery method is virtual. But it MUST return NameValueCollection, while the main serializing method `byte[]? SerializeProperties(JobDataMap data)` is private. So I cannot simply inherit from PostgreSQLDelegate and override only the serialization/deserialization methods :(

So once again - it works perfectly with that hacky setup via intermediate Dictionary. It is stored in database, restored from it and applied to job instances just fine.

**Comment by lahma** [comment, marked as answer]
`UseProperties = true` is protection to ensure data doesn't suffer from versioning problems, like storing a `Person` instance into database. If you don't need it and know that you won't have trouble with the types you use, why not just set it to false?

**Reply by justmara** [reply]
Now I feel myself a bit stupid: I were sure, that this flag enables `property injection` from JobData to Job properties. Don't know where this misunderstanding comes from. Now digging trough code I see that it really is some kind of check for AdoStore only and has nothing to do with `PropertySettingJobFactory`. My bad then :) Thank you

**Reply by lahma** [reply]
No worries, glad that I could help 🙂
